distributed_training/validator/forward.py

```diff
@@ -73,252 +73,269 @@ async def forward(self):
-                gradient_averaging_step.cancel()
-                bt.logging.info("Gradient Step Cancelled")
-                with self.grad_averager.use_averaged_gradients():
-                    self.opt.zero_grad()
-                bt.logging.info("Optimizer Gradients Zeroed")
-
-            self.step_scheduled = False
-        # Process the Train query responses
-        else:
-            bt.logging.info(
-                "Received Responses: "
-                + str(
+                bt.logging.info(
+                    "Received Responses: "
+                    + str(
+                        [
+                            {
+                                "Loss": response.loss,
+                                "Dataset Indices": (
+                                    min(response.dataset_indices),
+                                    max(response.dataset_indices),
+                                ),
+                                "IP": self.metagraph.axons[uid].ip,
+                                "Port": self.metagraph.axons[uid].port,
+                                "Hotkey": self.metagraph.axons[uid].hotkey,
+                            }
+                            for response, uid in zip(responses[0], self.miner_uids)
+                            if response.dendrite.status_code == 200
+                            and (response.dataset_indices is not None)
+                        ]
+                    )
+                )
+                self.average_loss = np.array(
                     [
-                        {
-                            "Loss": response.loss,
-                            "Dataset Indices": (
-                                min(response.dataset_indices),
-                                max(response.dataset_indices),
-                            ),
-                            "IP": self.metagraph.axons[uid].ip,
-                            "Port": self.metagraph.axons[uid].port,
-                            "Hotkey": self.metagraph.axons[uid].hotkey,
-                        }
+                        response.loss
                         for response, uid in zip(responses[0], self.miner_uids)
                         if response.dendrite.status_code == 200
                         and (response.dataset_indices is not None)
                     ]
-                )
-            )
-            self.average_loss = np.array(
-                [
-                    response.loss
-                    for response, uid in zip(responses[0], self.miner_uids)
-                    if response.dendrite.status_code == 200
-                    and (response.dataset_indices is not None)
-                ]
-            ).mean()
-            bt.logging.info(f"Current Average Miner Loss: {self.average_loss}")
+                ).mean()
+                bt.logging.info(f"Current Average Miner Loss: {self.average_loss}")
+
+    else:
+        time.sleep(self.all_reduce_timeout + 30)
```

Comment:
@mikkelmedm this is effectively doing what you where suggesting in our chat. For non-master validators they just wait for the duration of the all_reduce timeout and don't send any synapses to miners. To maximise the chances that all miners will respond to the master uids all_reduce call.